x-pack/plugins/monitoring/server/kibana_monitoring/lib/type_collector.js

```diff
@@ -47,15 +52,13 @@ export class TypeCollector {
   }
 
   /*
-   * @param {Object} collectorOptions (required)
-   * Fields:
-   * - type {String}
-   * - init {Function} (optional)
-   * - fetch {Function}
-   * - cleanup {Function} (optional)
+   * @param {String} type.type
```

Comment:
I wonder if we should make this an object parameter that makes these fields clearer? That or make `type` a class that users are expected to implement to take advantage of?

```js
register({ type, fetch, init, cleanup }) {
```

I would personally vote for the class because it makes it easier to require specific methods with the less opportunity for misunderstanding their purpose.

Reply:
I like the class implementation approach too, it's what I've been following for my pluggable things, like embeddables and pluggable panel actions.  Eventually when we move to typescript we'll be able to use their interfaces.  But this is fine too, if you want to leave as is.

Reply:
I had the same thought :D , but planned to work that out in a later PR: https://github.com/elastic/kibana/pull/18894/commits/60ba19566d1552c8b2e34bffd5ed9a00b911d7cf